### PR TITLE
Add `priority` field in lead

### LINKED ISSUE #906: Lead: Add low/medium/high priority in lead

### PR DESCRIPTION
**Pull request by frozenhelium**
Branched off from: https://github.com/the-deep/server/pull/472

Closes https://github.com/the-deep/deeper/issues/906

### CODE REVIEW

File: apps/lead/tests/test_apis.py [hunk 1]
```diff
@@ -81,6 +81,38 @@ def test_create_lead(self, assignee=None):
...
+        self.assert_201(response)
+
+        self.assertEqual(Lead.objects.count(), lead_count + 1)
+        r_data = response.json()
+        self.assertEqual(r_data['title'], data['title'])
+        self.assertEqual(r_data['assignee'], self.user.id)
+
+        # low is default priority
+        self.assertEqual(r_data['priority'], Lead.HIGH)
+
```
Comment (thenav56) [inline apps/lead/tests/test_apis.py:115]: @frozenhelium Please add separate tests for sorting and filter too. Thanks

Comment (frozenhelium) [reply apps/lead/tests/test_apis.py:115]: done

File: apps/lead/filter_set.py [hunk 2]
```diff
@@ -174,6 +179,22 @@ def ordering_filter(self, qs, name, value):
...
+                    default=models.Value(1),
+                    output_field=models.IntegerField(),)
+                ).order_by('priority_order')
+            elif ordering == '-priority':
+                qs = qs.annotate(priority_order=models.Case(
+                    models.When(priority=Lead.HIGH, then=models.Value(0)),
+                    models.When(priority=Lead.MEDIUM, then=models.Value(1)),
+                    models.When(priority=Lead.LOW, then=models.Value(2)),
+                    default=models.Value(1),
+                    output_field=models.IntegerField(),)
```
Comment (bewakes) [inline apps/lead/filter_set.py]: It's better to reuse the annotation. It's the same in both ordering values.

**Review by bewakes — CHANGES_REQUESTED** [review]

File: apps/lead/filter_set.py (hunk 2 above)
Comment (thenav56) [reply apps/lead/filter_set.py]: We are now using integerfield to store `priority`. So, annotation is removed. cc: @frozenhelium

**Review by thenav56 — APPROVED** [review]

**Pull request merged**